SpatialGDK/Source/SpatialGDK/Private/LoadBalancing/InterestManager.cpp

```diff
@@ -0,0 +1,615 @@
+#include "LoadBalancing/InterestManager.h"
+#include "Schema/ChangeInterest.h"
+#include "Utils/InterestFactory.h"
+
+// Try to grab improbable's phTree thingy
+
+#include <algorithm>
+#include <cmath>
+
+#include <immintrin.h>
+
+DECLARE_CYCLE_STAT(TEXT("InterestManagerCompute"), STAT_InterestManagerComputation, STATGROUP_SpatialNet);
+DECLARE_CYCLE_STAT(TEXT("InterestManagerComputeBroadphase"), STAT_InterestManagerComputationBroadphase, STATGROUP_SpatialNet);
+DECLARE_CYCLE_STAT(TEXT("InterestManagerComputeBox"), STAT_InterestManagerComputationBox, STATGROUP_SpatialNet);
+DECLARE_CYCLE_STAT(TEXT("InterestManagerComputeBoxSSE"), STAT_InterestManagerComputationBoxSSE, STATGROUP_SpatialNet);
+DECLARE_CYCLE_STAT(TEXT("InterestManagerComputeSort"), STAT_InterestManagerComputationSort, STATGROUP_SpatialNet);
+
+namespace SpatialGDK
+{
+FTagComponentStorage::FTagComponentStorage(Worker_ComponentId InComponentId)
+	: TagComponentId(InComponentId)
+{
+	Components.Add(TagComponentId);
+}
+
+void FTagComponentStorage::OnComponentAdded(Worker_EntityId EntityId, Worker_ComponentId ComponentId, Schema_ComponentData* Data)
+{
+	if (ensureAlways(TagComponentId == ComponentId))
+	{
+		TaggedObjects.Add(EntityId);
+		Modified.Add(EntityId);
+	}
+}
+
+void FTagComponentStorage::OnRemoved(Worker_EntityId EntityId)
+{
+	TaggedObjects.Remove(EntityId);
+	Modified.Remove(EntityId);
+}
+
+void FTagComponentStorage::OnUpdate(Worker_EntityId EntityId, Worker_ComponentId InComponentId, Schema_ComponentUpdate* Update)
+{
+	if (!ensure(false))
```

Comment:
?

Reply:
Tag components aren't intended to have any data on them, if they did they wouldn't be tags. So receiving an update would indicate somethings gone wrong.

Reply:
would suggest using ensuref with a message to that effect.